docs/LoweringCode.md

```diff
@@ -0,0 +1,265 @@
+
+The usage of this builder will become clearer with our same example, setting an array to value zero. This is the same example as in the prior section.
+``` C++
+// Defined values 0 and a 2 dimensional array with dim ub0 and ub1
+Value zero, array, ub0, ub1;
+
+// Define the krnl builder.
+KrnlBuilder createKrnl(rewriter, loc);
+
+// Define a 2-dimensional iteration space.
+ValueRange loopDef = createKrnl.defineLoops(2);
+
+// Create the loop.
+createKrnl.iterate(loopDef, loopDef, {zero, zero}, {ub0, ub1},
+  [&](KrnlBuilder &createKrnl, ValueRange loopInd){
+    // Loop body.
+    createKrnl.store(zero, array, loopInd);
+  });
+```
+**Code example 2: Zeroing an array using the new builder interface**
+
+Using this new scheme, we first define the 2D loop iteration space and then create the loop iteration structure using the `iterate` method. Since the loop is unoptimized, the same `loopDef` value range is passed as the first 2 parameters. The bounds are passed as 2 set of ordered values.
+
+Note that the lambda function create an `createKrnl` builder that is similar to that of the external environment (outside the loop), but customized for inside the loop. Se we can continue to use this overloaded builder to continue constructing krnl operations. In our case, we simply use the `loopInd` (2nd parameter of the lambda function), which are the current loop induction values, to define the element of the array that is set to zero.
+
+Some of the other operations that are often used are listed below.
+``` C++
+struct KrnlBuilder : public DialectBuilder {
+  // in addition to above...
+
+  // Memory operations.
+  Value load(Value memref, ValueRange indices = {});
+  void store(Value val, Value memref, ValueRange indices = {});
+
+  // Loop optimizations.
+  ValueRange block(Value loop, int64_t blockSize);
+  void permute(ValueRange loops, ArrayRef<int64_t> map);
+
+  // Simple setter for entire arrays.
+  void memcpy(Value dest, Value src, Value size);
+  void memset(Value dest, Value val);
+};
+```
+** Additional Krnl ops supported by the Krnl builder interface. **
+
+Above, both the load and store operations are used to create Krnl memory load and store operations. They should be used instead of the MLIR Affine or Standard dialect operations.
+
+The `block` method takes one loop definition (one value extracted from the output of a `defineLoop` operation) and will split that loop definition into 2, where the first one iterates over blocks of the given side, and the second one iterates inside of a given block. The two loop definitions are returned by the `block` method as a value range containing the two split loops described above.
+
+The `permute` method takes a list of loop definitions are ensure that the iterate will generate code reflecting the permuted order.
```

Comment:
are ensure that the iterate will generate -> and ensures that the loops will iterate according to the permuted order ?

Reply:
yes, done